File: hub/src/light/effect/amplitude_effects/basic_amplitude_modifier.py
from light.effect.resolved_effect import ResolvedEffect
from light.effect.partial_effect import PartialEffect
from light.effect.effect_priority import EffectPriority

class BasicAmplitudeModifier(EffectModifier):
    def __init__(self, amplitude):
        super().__init__(0)
        self.amplitude = amplitude

    def getEffect(self, t):
        return ResolvedEffect.createAmplitude(self.amplitude)

    def isTemporal(self):
        return False

    def isComplete(self, t):
        return False

File: hub/drumlite-js/src/light/effect/amplitude/BasicAmplitudeModifier.ts
import EffectModifier from "../EffectModifier"
import ResolvedEffect from "../../../effect/ResolvedEffect";
import { EffectParameters, EffectParameter } from "../PartialEffect";

export class BasicAmplitudeModifierParams extends EffectParameters {
    effectName = "Basic Amplitude Modifier";
    typeName = "Modifier";

    constructor(amplitude = 1) {
        super(0);
        this.params.amplitude = new EffectParameter<number>("Amplitude", amplitude);
    }
}

export default class BasicAmplitudeModifier extends EffectModifier<BasicAmplitudeModifierParams> {
    constructor(params: BasicAmplitudeModifierParams) {
        super(params);
    }

    public getEffect(t: number) {
        return ResolvedEffect.createAmplitude(this.params.params.amplitude.val as number);
    }

    public isTemporal() {
        return false;
    }

    public isComplete(t: number) {
        return false;
    }
}
